src/Elements/Input.react.js

```diff
@@ -176,12 +176,24 @@ class Input extends Component {
 
     _onChange(event) {
         const { value } = this.state;
-        const { max, min, onChange } = this.props;
+        const { max, min, onChange, required } = this.props;
         const type = this._getType();
         let newValue = event.target.value;
 
-        if (type === 'number' && _.isNumber(max) || _.isNumber(min)) {
-            newValue = newValue > max || newValue < min ? value : newValue;
+        if (type === 'number') {
+            if (_.isEmpty(newValue)) {
+                if (required) {
+                    newValue = _.isNumber(min) ? min : (_.isNumber(max)? max : 0);
+                }
+            } else {
+                newValue = +newValue;
+                if (_.isNumber(max)) {
+                    newValue = Math.min(max, newValue);
+                }
+                if (_.isNumber(min)) {
+                    newValue = Math.max(min, newValue);
+                }
+            }
```

Comment:
MUCH cleaner!  Thanks! :+1:   It's actually reasonable to figure out what this is doing now (for a simpleton such as myself).

@morethanfire can you take a look at latest and greatest please and make sure you're okay with it as well?

Reply:
Looks like unfortunately a conflict has to be resolved before this can be merged.  I think it can be resolved by taking this version as-is; it accomplishes the same as the fix introduced in 5b8e75b as well as introducing correct logic for the new behavior around the `required` prop.